### DISCUSSION TITLE
Question: how to remove semicolons?

### DISCUSSION THREAD

**Discussion by ClementNerma** [rome/tools#3469] [Q&A] [answered]
Hi there!

I'm currently starting up a new project with Rome, but I can't manage to find the option to automatically remove semicolons.

Even when I remove them manually the Rome formatter adds them back, which is not something I want. Prettier has a rule for this (`{ semi: false }`) but I don't see where this can be configured in Rome.

Could you please show me where the configuration option is?

Thanks in advance for your help§

**Comment by ematipico** [comment]
The team decided to not support this configuration for the time being. Here's the relevant discussion: https://github.com/rome/tools/discussions/2465#discussioncomment-2592369

**Comment by Lokua** [comment, marked as answer]
Looks like this is supported, now?
https://github.com/rome/tools/pull/3702

At this time is not published in the stable version of rome, though.

I'm totally new this tool and not sure if merges into main result in a new published npm package? I assumed `rome@next` or `rome@nightly` would but I tried both and still can't figure out how to actually configure it (I tried `"formatter": { "semicolons": "as-needed" }` in rome.json, also tried putting it into rules.

Edit: semicolons option does work for the CLI, I just can't figure out how/where to include it in rome.json

Edit 2: It's under javascript.formatter.semicolons. Sweet!

**Reply by Conaclos** [reply]
Yes, this requires `rome@nightly`. This is an option specific to JavaScript formatter (should be under `javascript.formatter`). Moreover, the option is in lowerCamelCcase `asNeeded`. See [here](https://github.com/rome/tools/issues/3535) for more details.
```json
{
  "javascript": {
    "formatter": {
      "semicolons": "asNeeded"
    }
  }
}
```
